scripts/bumpVersions.js

```diff
@@ -295,6 +306,26 @@ class VersionManager {
         this.addReleasedPackage(dep, bump);
       }
     }
+
+    // Bump anything that has this as a dep by a patch, all the way up the tree
+    if (status === 'released') {
+      for (let p in this.workspacePackages) {
+        if (this.workspacePackages[p].workspaceDependencies.includes(pkg)) {
+          if (this.releasedPackages.has(p)) {
+            continue;
+          }
+          let filePath = this.workspacePackages[p].location + '/package.json';
+          let pkg = JSON.parse(fs.readFileSync(filePath, 'utf8'));
+          if (pkg.private) {
+            continue;
+          }
+          let version = semver.parse(pkg.version);
+          let prerelease = version.prerelease;
+          let b = prerelease.length === 0 ? 'patch' : prerelease[0];
+          this.addReleasedPackage(p, b);
+        }
+      }
+    }
```

Comment:
Lemme just make sure I understand the logic here:
We iterate through every package (p) in our repo and do the following checks:
- Does said package (p) include the current bumped package as a dep? If yes, is the package (p) already in our list of packages to be released? Move to the next  if yes otherwise go to next bullet point.
-  Is package (p) private? Early break if yes other wise go to the next bullet point.
- Bump the package and add it to the list of packages to be released. If it has already been released, bump it appropriately. Otherwise, if the package is a prerelease (alpha, beta, rc) provide that info to `addReleasedPackage` so it bump everything that depends on it due to how we pin prerelease dependencies rather than using a caret.

> Otherwise, if the package is a prerelease (alpha, beta, rc) provide that info to `addReleasedPackage` so it bumps everything that depends on it due to how we pin prerelease dependencies rather than using a caret.

With regards to the above, that is the same bump strategy as [here](https://github.com/adobe/react-spectrum/blob/main/scripts/bumpVersions.js#L272-L290)? Can we get rid of the overall wrapping if statement there and then simplify the change in this PR to just have the private package check -> call? Haven't tried this yet, just noted how similar the logic was and was wondering if we could tweak it a bit without duplicating the logic

Reply:
yes, that's the logic

it's not quite the same as the bump strategy you shared with the link and since it's going in the opposite direction (up the tree instead of down) i felt it was best to keep them separate